internal/tests/api/scopes/scope_test.go

```diff
@@ -152,3 +155,102 @@ func TestErrors(t *testing.T) {
 	assert.NotNil(apiErr)
 	assert.EqualValues(http.StatusBadRequest, apiErr.Response().StatusCode())
 }
+
+func TestKeyDestruction(t *testing.T) {
+	ctx := context.Background()
+	tc := controller.NewTestController(t, &controller.TestControllerOpts{SchedulerRunJobInterval: time.Second})
+	t.Cleanup(tc.Shutdown)
+	c := tc.Client()
+	c.SetToken(tc.Token().Token)
+	sc := scopes.NewClient(c)
+
+	keys, err := sc.ListKeys(ctx, "global")
+	require.NoError(t, err)
+	assert.Len(t, keys.Items, 7)
+	for _, key := range keys.Items {
+		assert.Len(t, key.Versions, 1)
+	}
+
+	_, err = sc.RotateKeys(ctx, "global", false)
+	require.NoError(t, err)
+
+	keys, err = sc.ListKeys(ctx, "global")
+	require.NoError(t, err)
+	assert.Len(t, keys.Items, 7)
+	for _, key := range keys.Items {
+		assert.Len(t, key.Versions, 2)
+	}
+
+	// Root key is always last, by virtue of sorting by ID
+	rootKeyVersion := keys.Items[len(keys.Items)-1].Versions[1]
+	var destroyKeyVersion *scopes.KeyVersion
+	for _, key := range keys.Items {
+		if key.Purpose == kms.KeyPurposeDatabase.String() {
+			destroyKeyVersion = key.Versions[1]
+		}
+	}
+
+	jobs, err := sc.ListKeyVersionDestructionJobs(ctx, "global")
+	require.NoError(t, err)
+	assert.Len(t, jobs.Items, 0)
+
+	result, err := sc.DestroyKeyVersion(ctx, "global", rootKeyVersion.Id)
+	require.NoError(t, err)
+	assert.Equal(t, "completed", result.State)
+
+	jobs, err = sc.ListKeyVersionDestructionJobs(ctx, "global")
+	require.NoError(t, err)
+	assert.Len(t, jobs.Items, 0)
+
+	keys, err = sc.ListKeys(ctx, "global")
+	require.NoError(t, err)
+	assert.Len(t, keys.Items, 7)
+	for _, key := range keys.Items {
+		switch key.Purpose {
+		case "rootKey":
+			assert.Len(t, key.Versions, 1)
+		default:
+			assert.Len(t, key.Versions, 2)
+		}
+	}
+
+	result, err = sc.DestroyKeyVersion(ctx, "global", destroyKeyVersion.Id)
+	require.NoError(t, err)
+	assert.Equal(t, "pending", result.State)
+
+	jobs, err = sc.ListKeyVersionDestructionJobs(ctx, "global")
+	require.NoError(t, err)
+	assert.Len(t, jobs.Items, 1)
```

Comment:
if you didn't insert any data for rewrapping, wouldn't destroy return a completed status? per [this logic from destroy key version](https://github.com/hashicorp/boundary/pull/2498/files#diff-65fa95abdd78accdc6b1f3ac98751fe4700c8abc1ee696b8d6ae7e5403af0831R511-R517)?

Reply:
It would, yep! But since there's default data it doesn't.